pkg/controller/sparkapplication/submission_cache.go

```diff
@@ -0,0 +1,66 @@
+package sparkapplication
+
+/*
+A simple cache implementation that stores keys with a time-to-live (TTL) value.
+*/
+
+import (
+	"sync"
+	"time"
+)
+
+type item struct {
+	expiration int64
+}
+
+type SubmissionCache struct {
+	ttl   time.Duration
+	items map[string]*item
+	mu    sync.RWMutex
+}
+
+func NewSubmissionCache(ttl time.Duration) *SubmissionCache {
+	c := &SubmissionCache{
+		ttl:   ttl,
+		items: make(map[string]*item),
+	}
+
+	go c.startEviction()
+
+	return c
+}
+
+func (c *SubmissionCache) Set(key string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.items[key] = &item{
+		expiration: time.Now().Add(c.ttl).UnixNano(),
+	}
+}
+
+func (c *SubmissionCache) Exist(key string) bool {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	item, found := c.items[key]
+	if !found || item.expiration < time.Now().UnixNano() {
+		return false
+	}
+
+	return true
+}
+
+func (c *SubmissionCache) startEviction() {
+	for {
+		time.Sleep(c.ttl)
```

Comment:
not sure to understand,
there is no stop the world here if i'm correct.
`startEviction` is call in a goroutine by `NewSubmissionCache`

Reply:
it will delete items from the cache every 5 minutes, so in the worst-case scenario, an item in the cache can stay for ~10 minutes, right? If a customer submits a lot of apps, the cache can grow significantly. We don't store a lot of data in the cache, only the string key, so it shouldn't cause any performance issues, in theory. But cleaning the cache more smoothly does make sense, imo, it won't allow the cache to grow fast.

Reply:
just understand the meaning of "stop the world" now, 
agree with your point 👍 

let's iterate each 30 seconds